constellations/gemini-admin/src/emailPayment/accountInfo.js

```diff
@@ -0,0 +1,151 @@
+"use strict";
+require("../account/accountInfo.sass");
+import React from "react";
+import { Link } from "react-router-dom";
+import { getFullName } from "../utils/userUtils.js";
+
+export class AccountInfo extends React.Component {
+    state = {
+        selectedUsers: [],
+        selectedUserEmails: [],
+    };
+
+    formatCurrency = (amount) => {
+        return new Intl.NumberFormat("en-US", {
+            style: "currency",
+            currency: "USD",
+        }).format(amount);
+    };
+
+    onCheckUser = (e, userId) => {
+        const users = this.props.users;
+        if (e.target.checked) {
+            var emails = this.state.selectedUserEmails;
+            var currentSelectedUsers = this.state.selectedUsers;
+            const checkedUser = users.find((user) => user.id == userId);
+            emails.push(checkedUser.email);
+            currentSelectedUsers.push(checkedUser);
+            this.setState({
+                selectedUserEmails: emails,
+                selectedUsers: currentSelectedUsers,
+            });
+        } else {
+            var emails = this.state.selectedUserEmails;
+            var currentSelectedUsers = this.state.selectedUsers;
+            const uncheckedUser = users.find((user) => user.id == userId);
+            emails.splice(emails.indexOf(uncheckedUser.email), 1);
+            currentSelectedUsers.splice(
+                currentSelectedUsers.indexOf(uncheckedUser),
+                1
+            );
+            this.setState({
+                selectedUserEmails: emails,
+                selectedUsers: currentSelectedUsers,
+            });
+        }
+    };
+
+    checkUserExists = () => {
+        const users = this.props.users;
+    };
+
+    render = () => {
+        const id = this.props.id;
+        const accountEmail = this.props.email;
+        const users = this.props.users;
+        const transactions = this.props.transactions;
+        const name = this.props.name;
+
+        var emails = [];
+        emails = users.map((user) => user.email);
+
+        const userRows = users.map((user, index) => {
+            var status = user.isGuardian ? "(guardian" : "(student";
+            status += user.email == accountEmail ? ", primary contact)" : ")";
+
+            if (this.state.selectedUsers.length > 0) {
+                for (var i = 0; i < this.state.selectedUsers.length; i++) {
+                    if (
+                        users.find(
+                            (user) => user == this.state.selectedUsers[i]
+                        ) == undefined
+                    ) {
+                        this.setState({
```

Comment:
This can be a bit cleaner if:

```
var user = users.find(...)
if (!user) {
   ...
}
```

In Javascript, there's this idea of "falsy" values meaning 0, false, undefined, NaN, "". All of these values can work with the `!` operator which will make them "truthy".

So !(undefined) evaluates to true